x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationIndexService.java

```diff
@@ -275,6 +289,7 @@ private void updateSearchApplication(SearchApplication app, boolean create, Acti
                     .field(SearchApplication.NAME_FIELD.getPreferredName(), app.name())
                     .field(SearchApplication.INDICES_FIELD.getPreferredName(), app.indices())
                     .field(SearchApplication.ANALYTICS_COLLECTION_NAME_FIELD.getPreferredName(), app.analyticsCollectionName())
+                    .field(SearchApplication.TEMPLATE_FIELD.getPreferredName(), app.searchTemplate())
```

Comment:
Questions surrounding this field: 

- Should we add validation to the put search application command that the template cannot be changed once it's set?
- Do we need to validate the template exists when we attach it to a search application?

We'll likely want to update the API contracts, including the asciidocs and yaml tests

Reply:
> I think we might want to be able to create a default search template, if one is not provided. For this reason, I have not added a validation yet.

I think it's a good idea to create a default if no template is specified, but what happens if they specify "template-that-does-not-exist?" Would we just take care of that with informative errors when they try to search?

Reply:
I don't think we have fully defined what the behaviour should be for default templates.
For now, we could say that setting the template to `null`, means reverting back to using the default.